pkg/sql/distsql_physical_planner.go

```diff
@@ -797,9 +838,20 @@ func (dsp *DistSQLPlanner) createTableReaders(
 		return physicalPlan{}, err
 	}
 
-	spanPartitions, err := dsp.partitionSpans(planCtx, n.spans)
-	if err != nil {
-		return physicalPlan{}, err
+	var spanPartitions []spanPartition
+	if n.hardLimit == 0 && n.softLimit == 0 {
+		spanPartitions, err = dsp.partitionSpans(planCtx, n.spans)
+		if err != nil {
+			return physicalPlan{}, err
+		}
+	} else {
```

Comment:
You might want to make this explicitly `else if n.hardLimit != 0` to make it clear to the reader what we're doing here. And you could include a panicing `else` case too, to make people who modify the planning code above remember to make any corresponding modifications to the spanPartition creation code here.